### ISSUE TITLE
wasmpack-init should respect CARGO_HOME environment variable

### ISSUE THREAD

**Issue by olalonde** [rustwasm/wasm-pack#814]
wasmpack-init currently uses the parent directory of rustup to determiner where it should install. It should instead default to the `CARGO_HOME` environment variable. 

https://github.com/rustwasm/wasm-pack/blob/master/src/installer.rs#L62

**Comment by alun** [comment]
I agree - running into an issue when I've overridden my `CARGO_HOME` and `CARGO_TARGET` for better caching while building inside transient docker container.

Looks like `wasm-pack` doesn't take them into account so it keeps spending a lot of time on rebuilding deps every single build.
